src/lib.rs

```diff
@@ -111,6 +111,12 @@ pub struct InstanceSettings {
     /// use the same as timeout when sending
     #[builder(default = "Duration::from_secs(10)")]
     request_timeout: Duration,
+    /// Maximum size of the HTTP response body in bytes.
+    /// This is limited to prevent potential memory exhaustion DoS attacks.
+    ///
+    /// Defaults to 2MB
+    #[builder(default = "2 * 1024 * 1024")]
+    response_body_size: usize,
```

Comment:
I don't think it is necessary to make the maximum response size configurable. The biggest data I see is about 5 KB, so a hardcoded limit of 100 KB would be totally sufficient.

Reply:
Removed the configuration option, the shim now has a 100KB limit hardcoded